packages/app-accounts/src/UploadButton.tsx

```diff
@@ -0,0 +1,237 @@
+// Copyright 2017-2018 @polkadot/app-accounts authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Button$Sizes } from '@polkadot/ui-app/Button/types';
+import { BareProps, I18nProps } from '@polkadot/ui-app/types';
+import { KeyringPair$Json } from '@polkadot/util-keyring/types';
+
+import React from 'react';
+import translate from './translate';
+import { Trans } from 'react-i18next';
+import isUndefined from '@polkadot/util/is/undefined';
+import keyring from '@polkadot/ui-keyring/index';
+
+import AddressMini from '@polkadot/ui-app/AddressMini';
+import Button from '@polkadot/ui-app/Button';
+import File from '@polkadot/ui-app/Params/Param/File';
+import Modal from '@polkadot/ui-app/Modal';
+import Unlock from '@polkadot/ui-signer/Unlock';
+
+type State = {
+  address: string,
+  password: string,
+  isPasswordModalOpen: boolean,
+  error?: React.ReactNode,
+  uploadedFileKeyringPair: KeyringPair$Json | undefined
+};
+
+type Props = I18nProps & BareProps & {
+  icon?: string,
+  isCircular?: boolean,
+  isPrimary?: boolean,
+  size?: Button$Sizes,
+  onChangeAccount: any
+};
+
+class UploadButton extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = this.emptyState();
+  }
+
+  handleUploadedFiles = (fileBytes: Uint8Array): void => {
+    const { t } = this.props;
+
+    if (!fileBytes || !fileBytes.length) {
+      console.error('Error retrieving file list');
+      return;
+    }
+
+    const blob = new Blob([fileBytes], { type: 'text/plain;charset=utf-8' });
+
+    const fileReader = new FileReader();
+
+    fileReader.onload = (e) => {
+      try {
+        if (!isUndefined(e) && e.target !== null) {
+          const fileContents: any = JSON.parse(e.target.result);
+
+          if (Object.keys(fileContents).includes('address' && 'encoding' && 'meta')) {
```

Comment:
`'address' && 'encoding' && 'meta'` evaluates to -- `meta`

So the logic is wrong.